### PR TITLE
doc: matter: update links to docset

### PR DESCRIPTION
**Pull request by greg-fer**
Replaced GitHub links with cross-links to Matter docset.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
Intersphinx needs to be configured for matter in nrf docset conf.py. Furthermore, a nrf dependency on matter has to be added to the docs CMakeLists.txt

File: doc/matter/conf.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 extensions = [
     "recommonmark",
+    "sphinx.ext.intersphinx",
```
Comment (gmarull) [inline doc/matter/conf.py:22]:
```suggestion
```

This is not required, `matter` doesn't reference to any other docset.  `nrf` does, though (see https://github.com/nrfconnect/sdk-nrf/blob/master/doc/nrf/conf.py#L81)

File: doc/nrf/ug_matter_configuring.rst [hunk 2]
```diff
@@ -89,7 +89,7 @@ For information about how to configure and use the required components, complete
 * Depending on the Matter controller type:
 
   * Python Matter controller - See :ref:`ug_matter_tools` for building instructions and `Working with Python Controller`_ in the Matter documentation for information about testing.
-  * Android Mobile controller - See :ref:`ug_matter_tools` for building instructions and `Commissioning nRF Connect Accessory using Android CHIPTool`_ in the Matter documentation for information about testing.
```
Comment (gmarull) [inline doc/nrf/ug_matter_configuring.rst:92]: this link needs to be removed from `docs/nrf/links.txt`.

File: doc/nrf/links.txt [hunk 3]
```diff
@@ -91,7 +91,6 @@
 .. _`SMP over Bluetooth`: https://github.com/apache/mynewt-mcumgr/blob/master/transport/smp-bluetooth.md
 
 .. _`Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/tree/ee7ed2a/src/android/CHIPTool
-.. _`Commissioning nRF Connect Accessory using Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/docs/guides/nrfconnect_android_commissioning.md
 .. _`Working with Python Controller`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/docs/guides/python_chip_controller_building.md
```
Comment (gmarull) [inline doc/nrf/links.txt:94]: may also be removed/updated?

File: doc/nrf/links.txt [hunk 4]
```diff
@@ -91,7 +91,6 @@
 .. _`SMP over Bluetooth`: https://github.com/apache/mynewt-mcumgr/blob/master/transport/smp-bluetooth.md
 
 .. _`Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/tree/ee7ed2a/src/android/CHIPTool
```
Comment (gmarull) [inline doc/nrf/links.txt:93]: ditto

File: doc/nrf/links.txt [hunk 5]
```diff
@@ -91,7 +91,6 @@
...
-.. _`Commissioning nRF Connect Accessory using Android CHIPTool`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/docs/guides/nrfconnect_android_commissioning.md
 .. _`Working with Python Controller`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/docs/guides/python_chip_controller_building.md
 .. _`Performing Device Firmware Upgrade in Matter device`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/docs/guides/nrfconnect_examples_software_update.md
 .. _`Matter Protocol Overview`: https://github.com/nrfconnect/sdk-connectedhomeip/blob/ee7ed2a/README.md
```
Comment (gmarull) [inline doc/nrf/links.txt:96]: ditto

File: doc/nrf/links.txt (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/links.txt:94]: Yes, I want to test two links first, as I cannot build locally yet (hasn't had time and last week had errors all the time).

File: doc/nrf/links.txt (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/links.txt:94]: @LuDuda , this links to the main page of the Matter repository, but the description there is heavily outdated, to quote some Matter members involved in re-writing this page. Can we remove this link from ug_matter_architectures.rst?

Comment (LuDuda) [reply doc/nrf/links.txt:94]: @greg-fer I'm fine with removing the sentence about this page in Matter architecture. Thanks.

**Review by LuDuda — APPROVED** [review]
Thanks :+1:

File: doc/nrf/ug_matter_configuring.rst [hunk 6]
```diff
@@ -88,8 +88,8 @@ For information about how to configure and use the required components, complete
 * Configuring the Thread Border Router on a Raspberry Pi - see :ref:`ug_thread_tools_tbr` in the |NCS| documentation
 * Depending on the Matter controller type:
 
-  * Python Matter controller - See :ref:`ug_matter_tools` for building instructions and `Working with Python Controller`_ in the Matter documentation for information about testing.
-  * Android Mobile controller - See :ref:`ug_matter_tools` for building instructions and `Commissioning nRF Connect Accessory using Android CHIPTool`_ in the Matter documentation for information about testing.
+  * Python Matter controller - See :ref:`ug_matter_tools` for building instructions and :doc:`python_chip_controller_building` in the Matter documentation for information about testing.
```
Comment (LuDuda) [inline doc/nrf/ug_matter_configuring.rst:91]:
```suggestion
  * Python Matter controller - See :ref:`ug_matter_tools` for building instructions and :doc:`matter:python_chip_controller_building` in the Matter documentation for information about testing.
```

File: doc/nrf/ug_matter_configuring.rst [hunk 7]
```diff
@@ -113,7 +113,7 @@ To use this setup, you need the following hardware:
 For information about how to configure and use the required components, see the following user guides:
 
 * Configuring Thread Border Router on a PC or a Raspberry Pi - see :ref:`ug_thread_tools_tbr` in the |NCS| documentation
-* Python Matter controller - See :ref:`ug_matter_tools` for building instructions and `Working with Python Controller`_ in the Matter documentation for information about testing.
+* Python Matter controller - See :ref:`ug_matter_tools` for building instructions and :doc:`python_chip_controller_building` in the Matter documentation for information about testing.
```
Comment (LuDuda) [inline doc/nrf/ug_matter_configuring.rst:116]:
```suggestion
* Python Matter controller - See :ref:`ug_matter_tools` for building instructions and :doc:`matter:python_chip_controller_building` in the Matter documentation for information about testing.
```

**Review by gmarull — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

**Pull request merged**